staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -620,3 +634,123 @@ func setTargetKind(obj Object, kind schema.GroupVersionKind) {
 	}
 	obj.GetObjectKind().SetGroupVersionKind(kind)
 }
+
+//  all specified versions must be in the same group, and the specified order overwrites any previously specified order for this group
+func (s *Scheme) SetVersionPriority(versions ...schema.GroupVersion) error {
+	groups := sets.String{}
+	order := []string{}
+	for _, version := range versions {
+		groups.Insert(version.Group)
+		order = append(order, version.Version)
+	}
+	if len(groups) != 1 {
```

Comment:
would also be useful to ensure an empty or internal version wasn't passed to this function